File: tests/lazy_evl.cpp
#include <iostream>

int print_a_returns_one()
{
    std::cout << "a" << std::endl;
    return 1;
}

int print_b_returns_nought()
{
    std::cout << "b" << std::endl;
    return 0;
}

int test() {
    int a = print_a_returns_one();
    int b = print_b_returns_nought();

    return b || a;
}

int main(void)
{
    std::cout << test() << std::endl;

    return 0;
}

File: tests/lazy_evl.py
def print_a_returns_one() -> bool:
    print("a")
    return True

def print_b_returns_nought() -> bool:
    print("b")
    return False

if print_a_returns_one() or print_b_returns_nought():
    print("done")
else:
    print("only first")
